src/org/rust/lang/inspections/ApproxConstantInspection.kt

```diff
@@ -0,0 +1,78 @@
+package org.rust.lang.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustLitExpr
+import org.rust.lang.core.psi.RustVisitor
+import java.util.*
+
+class ApproxConstantInspection : LocalInspectionTool() {
+
+    override fun getGroupDisplayName() = "Rust"
+    override fun getDisplayName() = "Approximate Constants"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustVisitor() {
+            override fun visitLitExpr(o: RustLitExpr) {
+                analyzeLiteral(o.floatLiteral ?: return, holder);
+            }
+        }
+    }
+
+    private fun analyzeLiteral(psiElement: PsiElement, holder: ProblemsHolder) {
+        val constant = KNOWN_CONSTS
+            .find { isApproxConst(it.value, psiElement.text, it.minDigits) } ?: return
+
+        holder.registerProblem(psiElement,
+            "Approximate value of `f{32,64}::${constant.name}` found. Consider using it directly.");
```

Comment:
There is no `f64::PI` in Rust, but there is `std::f64::consts::PI`. `f{32, f64}` is misleading, I think pure `f64` is clearer.